components/tidb_query/src/rpn_expr/impl_cast.rs

```diff
@@ -317,6 +341,139 @@ cast_as_unsigned_integer!(DateTime, cast_datetime_as_uint);
 cast_as_unsigned_integer!(Duration, cast_duration_as_uint);
 cast_as_unsigned_integer!(Json, cast_json_as_uint);
 
+/// The implementation for push down signature `CastIntAsJson` from unsigned integer.
+#[rpn_fn]
+#[inline]
+pub fn cast_uint_as_json(val: &Option<Int>) -> Result<Option<Json>> {
+    match val {
+        None => Ok(None),
+        Some(val) => Ok(Some(Json::U64(*val as u64))),
+    }
+}
+
+#[rpn_fn]
+#[inline]
+pub fn cast_int_as_json_boolean(val: &Option<Int>) -> Result<Option<Json>> {
+    match val {
+        None => Ok(None),
+        Some(val) => Ok(Some(Json::Boolean(*val != 0))),
+    }
+}
+
+#[rpn_fn(capture = [extra])]
+#[inline]
+pub fn cast_string_as_json(
+    extra: &RpnFnCallExtra<'_>,
+    val: &Option<Bytes>,
+) -> Result<Option<Json>> {
+    match val {
+        None => Ok(None),
+        Some(val) => {
+            if extra
+                .ret_field_type
+                .flag()
+                .contains(FieldTypeFlag::PARSE_TO_JSON)
+            {
+                let s = box_try!(String::from_utf8(val.to_owned()));
+                let val: Json = s.parse()?;
+                Ok(Some(val))
+            } else {
+                // FIXME: port `JSONBinary` from TiDB to adapt if the bytes is not a valid utf8 string
+                let val = unsafe { String::from_utf8_unchecked(val.to_owned()) };
+                Ok(Some(Json::String(val)))
+            }
+        }
+    }
+}
+
+/// The implementation for push down signature `CastIntAsDuration`
+#[rpn_fn(capture = [ctx, extra])]
+#[inline]
+pub fn cast_int_as_duration(
+    ctx: &mut EvalContext,
+    extra: &RpnFnCallExtra<'_>,
+    val: &Option<Int>,
+) -> Result<Option<Duration>> {
+    match val {
+        None => Ok(None),
+        Some(val) => {
+            let dur = Duration::from_i64(ctx, *val, extra.ret_field_type.get_decimal() as u8)?;
+            Ok(Some(dur))
+        }
+    }
+}
+
+#[rpn_fn(capture = [extra])]
+#[inline]
+pub fn cast_real_as_duration(
+    extra: &RpnFnCallExtra<'_>,
+    val: &Option<Real>,
+) -> Result<Option<Duration>> {
+    match val {
+        None => Ok(None),
+        Some(val) => {
+            // FIXME: the `Duration::parse` should return `Res` which indicates whether `Overflow` or `Truncated` occurred.
+            let dur = Duration::parse(
+                val.into_inner().to_string().as_bytes(),
+                extra.ret_field_type.get_decimal() as i8,
+            )?;
+            Ok(Some(dur))
+        }
+    }
+}
+
+#[rpn_fn(capture = [extra])]
+#[inline]
+pub fn cast_bytes_as_duration(
+    extra: &RpnFnCallExtra<'_>,
+    val: &Option<Json>,
+) -> Result<Option<Duration>> {
+    match val {
```

Comment:
Just a suggestion, I prefer to use `map_or` to prevent repeating myself:
```rust
val.map_or(Ok(None), |val| Ok(Some(Duration::parse(val.unquote()?.as_bytes(), ...)?)))
```